### PR TITLE
connect to mongodb + script to create DOC-proposed structure

### PR DESCRIPTION
**Pull request by cwong6854**
- `route.ts` handler to connect to database locally
- `seed_script.js` used to create DOC proposed structure of objects to seed into MongoDB
Screenshot 2024-11-23 at 6 03 24 PM

### CODE REVIEW

File: app/api/mongodb/seed_script.js [hunk 1]
```diff
@@ -0,0 +1,136 @@
+const fs = require("fs");
+const os = require('os');
+const path = require("path")
+
+let booksMapping = {
```
Comment (rogerrw) [inline app/api/mongodb/seed_script.js]: this should be a const. in general, use const whenever possible

File: app/api/mongodb/seed_script.js [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+                }
+            }
+        })
+
+    }
+    dirPaths.forEach((path) => {
+        readDirSync(path)
+
+    })
+    fs.writeFileSync("data.json", JSON.stringify(objs, null, 4), "utf8");
```
Comment (rogerrw) [inline app/api/mongodb/seed_script.js]: I think writing the file here is ok for the sake of verifying the results, but does this script update the mongo db anywhere?

Also, we shouldn't check in the resulting data.json. That should either be in the .gitignore or deleted

File: app/api/mongodb/data.json
Comment (rogerrw) [inline app/api/mongodb/data.json:1]: remove from commit

**Review by rogerrw — CHANGES_REQUESTED** [review]

File: app/api/mongodb/seed_script.js (hunk 2 above)
Comment (cwong6854) [reply app/api/mongodb/seed_script.js]: created .gitignore to hide data.json

**Review by cwong6854 — COMMENTED** [review]
able to seed mongodb and test results of bible data

File: package.json [hunk 3]
```diff
@@ -1,5 +1,6 @@
 {
   "private": true,
+  "type": "module",
   "scripts": {
```
Comment (rogerrw) [inline package.json]: `pnpm run seed`

"scripts": {
  "seed": "node app/api/mongodb/seed_script.js",
}

**Pull request merged**